src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/HubMessage.cs

```diff
@@ -11,39 +12,48 @@ protected HubMessage()
         {
         }
 
-        // Initialize with capacity 4 for the 2 built in protocols and 2 data encoders
-        private readonly List<SerializedMessage> _serializedMessages = new List<SerializedMessage>(4);
+        // Initialize with capacity 2 for the 2 built in protocols
+        private object _lock = new object();
+        private readonly List<SerializedMessage> _serializedMessages = new List<SerializedMessage>(2);
 
-        public byte[] WriteMessage(HubProtocolReaderWriter protocolReaderWriter)
+        public byte[] WriteMessage(IHubProtocol protocol)
         {
-            for (var i = 0; i < _serializedMessages.Count; i++)
+            // REVIEW: Revisit lock
```

Comment:
This lock is needed because multiple connections could be trying to serialize this message. It was missed in the initial PR that added it.